lib/widgets/input_field.dart

```diff
@@ -0,0 +1,97 @@
+import 'package:flutter/material.dart';
+import 'package:podcasty/models/input_appearance.dart';
+import 'package:podcasty/values/app_colors.dart';
+import 'package:podcasty/values/enums.dart';
+import 'package:podcasty/values/string_constants.dart';
+
+class InputField extends StatelessWidget {
+  final String hintText;
+  final IconData? suffixIcon;
+  final bool obscureText;
+  final TextInputType keyboardType;
+  final InputFieldAppearance appearance;
+  final Color? iconColor;
+  final TextInputAction? textInputAction;
+  final ValueSetter<String>? onSubmitted;
+  final VoidCallback? suffixIconPress;
+  final ValidationType validationType;
+
+  const InputField({
+    super.key,
+    required this.hintText,
+    this.suffixIcon,
+    this.obscureText = false,
+    this.keyboardType = TextInputType.text,
+    this.appearance = const InputFieldAppearance(),
+    this.iconColor = AppColors.black,
+    this.textInputAction = TextInputAction.next,
+    this.onSubmitted,
+    this.suffixIconPress,
+    this.validationType = ValidationType.name,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    String? validateInput(String? value) {
+      value = value?.trim(); // Trim the value
+      if (value?.isEmpty ?? true) {
+        return AppStrings.emptyError;
+      }
+      return validationType.validate(value); // Validate the trimmed value
+    }
```

Comment:
Please never create a function within another function. Please move it to below build method.

Reply:
Resolved